### PR TITLE
Added warning about where instantiate IpregistryClient and how set API key after that

### PR DESCRIPTION
**Pull request by matiaslopezd**
I changed the `tryout` to `REPLACE_WITH_API_KEY`, which is more readable. Also in Caching, I added a warning where instantiate `IpregistryClient` related to cache behavior. Finally, I believe is useful to mention how to set an API Key after instantiating a new `IpregistryClient` to reuse cache with all API Keys.

😄

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -34,7 +34,7 @@ This is a very simple example. This creates a Ipregistry client and retrieves IP
 ```javascript
 const {IpregistryClient} = require('@ipregistry/client');
 
-const client = new IpregistryClient('tryout');
+const client = new IpregistryClient('REPLACE_WITH_API_KEY');
```
Comment (lpellegr) [inline README.md:37]: Thanks for the suggestion. I have pushed a commit related to this change since I prefer to use `YOUR_API_KEY` instead of `REPLACE_WITH_API_KEY`.

File: README.md [hunk 2]
```diff
@@ -109,6 +109,20 @@ Browser support:
...
+
+```js
+const {IpregistryClient} = require('@ipregistry/client');
+const client = new IpregistryClient('secret', cache);
+
+// Here
+client.config.apiKey = 'REPLACE_WITH_API_KEY';
+```
+
+This is useful when you need reuse the [cache](https://github.com/ipregistry/ipregistry-javascript#caching) with all API Key you created.
```
Comment (lpellegr) [inline README.md:124]: What's the business use case? I mean why is a single API key not enough? the only case I can think of is if you try to combine API keys from multiple free accounts but this is against the terms of the service.

File: README.md [hunk 3]
```diff
@@ -118,26 +132,28 @@ The Ipregistry client library has built-in support for in-memory caching.
 By default caching is disabled. Below are examples to enable and configure a caching strategy.
 Once enabled, the default cache implementation memoizes for 10min the most 2048 recently used lookups on server side (16 when used in a browser).
 
+**WARNING:** You need instantiate `IpregistryClient` outside of a requests or middlewares. If you instantiate, inside on every request, the behavior of caching not will as you expect.
```
Comment (lpellegr) [inline README.md:135]: I guess you mean in the case the client is used in a Function as a service such as Firebase Function, Google Cloud Function, or AWS Lambda?

**Review by lpellegr — COMMENTED** [review]
Thanks for the PR. I have added some comments to ask for clarification.

File: README.md (hunk 2 above)
Comment (matiaslopezd) [reply README.md:124]: > What's the business use case? I mean why is a single API key not enough? the only case I can think of is if you try to combine API keys from multiple free accounts but this is against the terms of the service.

In our case, we want to use two API keys for different endpoints and rate-limiting, but in the same account. Also, we will understand which service generates more request to IPRegistry :) With that case of use, we instantiate an `IpregistryClient` on the top and reuse cache on the two endpoints.

**In summary, take advantage of the multiple API Key options to monitor consumption per endpoint.** (In fact, you're the only provider that offers this, is so cool and useful).

I understand your point about the abuse of your service, but that case could happen with this method or instantiate multiples `IpregistryClient` with different API Keys (don't do it 🤨).

Feel free to omit this suggestion if you believe can avoid abuse of your service.

File: README.md (hunk 3 above)
Comment (matiaslopezd) [reply README.md:135]: Also, but not necessarily in those cases. Instantiate cache in the wrong place can provoke get out of memory.

Ex.
```js
// Bad
app.get('/my-endpoint', function (req, res) {
  const client = new IpregistryClient('REPLACE_WITH_API_KEY'); // <--- Wrong
  await client.lookup(...);
});

// Good
const client = new IpregistryClient('REPLACE_WITH_API_KEY');

app.get('/my-endpoint', function (req, res) {
  await client.lookup(...);
});
```

Can be obvious, but in the major cache libraries for Node, developers use it in the wrong place and send like an issue.

File: README.md (hunk 2 above)
Comment (lpellegr) [reply README.md:124]: Thanks for the clarification, I better understand your use case: using the same client cache for different API keys.

**Comment by lpellegr** [comment]
Thanks again for this PR. We will not directly merge your PR but I will consider your changes by writing things a bit differently.

**Comment by lpellegr** [comment]
Done:

- https://github.com/ipregistry/ipregistry-javascript/commit/a246b3b19e318d2864a4742c073f87206499fd5c
- https://github.com/ipregistry/ipregistry-javascript/commit/08a0af7a32367d2247fdd0e0f2ebd4a0f3bf8489
